symengine/ntheory.cpp

```diff
@@ -1498,4 +1498,12 @@ int mobius(const Integer &a)
     }
 }
 
+int mertens(const unsigned long a)
+{
+    unsigned long mertens = 0;
```

Comment:
Your function returns `int` but inside, you have returned an `unsigned long`, the test cases passes only because the answer is `int` casted.